include/maidsafe/crux/detail/multiplexer.hpp

```diff
@@ -137,7 +137,8 @@ class multiplexer : public std::enable_shared_from_this<multiplexer>
     using socket_map = std::map<endpoint_type, socket_base *>;
     socket_map sockets;
 
-    std::atomic<int> receive_calls;
+    using receive_calls_underlying_type = int;
+    std::atomic<receive_calls_underlying_type> receive_calls;
```

Comment:
Yes.  It's unrelated to the `decltype` in the cast, even this emits a warning:

    std::int64_t i(0);
    static_cast<std::atomic<int>>(i);